src/Sarif.UnitTests/Visitors/RebaseUriVisitorTests.cs

```diff
@@ -0,0 +1,207 @@
+            baseUriDictionary[srcRoot].ShouldBeEquivalentTo(srcRootUri);
+            baseUriDictionary.Should().ContainKey(bldRoot);
+            baseUriDictionary[bldRoot].ShouldBeEquivalentTo(bldRootUri);
+        }
+
+        [Fact]
+        public void RebaseUriVisitor_VisitRun_ReplacesBaseUriDictionaryWhenIncorrect()
+        {
+            // Slightly roundabout.  We want to randomly test this, but we also want to be able to repeat this if the test fails.
+            int randomSeed = (new Random()).Next();
+            Random random = new Random(randomSeed);
+
+            Run oldRun = RandomSarifLogGenerator.GenerateRandomRun(random);
+            oldRun.Properties = new Dictionary<string, SerializedPropertyInfo>();
+            SerializedPropertyInfo oldData = new SerializedPropertyInfo("42", false);
+            oldRun.Properties.Add(RebaseUriVisitor.BaseUriDictionaryName, oldData);
+
+            RebaseUriVisitor rebaseUriVisitor = new RebaseUriVisitor("SRCROOT", new Uri(@"C:\src\root"));
+            
+            Run newRun = rebaseUriVisitor.VisitRun(oldRun);
+
+            newRun.Properties.Should().ContainKey(RebaseUriVisitor.BaseUriDictionaryName);
+
+            Dictionary<string, Uri> baseUriDictionary = RebaseUriVisitor.DeserializePropertyDictionary(newRun.Properties[RebaseUriVisitor.BaseUriDictionaryName]);
+
+            baseUriDictionary.Should().ContainKey("SRCROOT");
+            baseUriDictionary["SRCROOT"].ShouldBeEquivalentTo(new Uri(@"C:\src\root"));
+
+            newRun.Properties.Should().ContainKey(RebaseUriVisitor.BaseUriDictionaryName + RebaseUriVisitor.IncorrectlyFormattedDictionarySuffix);
+            newRun.Properties[RebaseUriVisitor.BaseUriDictionaryName + RebaseUriVisitor.IncorrectlyFormattedDictionarySuffix].ShouldBeEquivalentTo(oldData);
+        }
+
+        [Fact]
+        public void RebaseUriVisitor_VisitRun_CorrectlyPatchesFileDictionaryKeys()
+        {
+            // Slightly roundabout.  We want to randomly test this, but we also want to be able to repeat this if the test fails.
+            int randomSeed = (new Random()).Next();
+            Random random = new Random(randomSeed);
+
+            Run oldRun = RandomSarifLogGenerator.GenerateRandomRun(random);
+            
+            RebaseUriVisitor rebaseUriVisitor = new RebaseUriVisitor("SRCROOT", new Uri(@"C:\src\"));
+
+            Run newRun = rebaseUriVisitor.VisitRun(oldRun);
+
+            newRun.Properties.Should().ContainKey(RebaseUriVisitor.BaseUriDictionaryName);
+            
+            newRun.Files.Keys.Where(k => k.StartsWith(@"C:\src\")).Should().BeEmpty();
+        }
+
+        [Fact]
+        public void RebaseUriVisitor_VisitRun_DoesNotPatchFileDictionaryKeysWhenNotABaseUri()
+        {
+            // Slightly roundabout.  We want to randomly test this, but we also want to be able to repeat this if the test fails.
+            int randomSeed = (new Random()).Next();
```

Comment:
if you provide randomization, it might be helpful to ensure that random seed is persisted to the log file. This seed may be required in order to reproduce a specific failure.